odoo/models.py

```diff
@@ -4321,20 +4323,23 @@ def _check_company(self, fnames=None):
 
         if inconsistencies:
             lines = [_("Incompatible companies on records:")]
-            company_msg = _lt("- Record is company “%(company)s” and “%(field)s” (%(fname)s: %(values)s) belongs to another company.")
-            record_msg = _lt("- “%(record)s” belongs to company “%(company)s” and “%(field)s” (%(fname)s: %(values)s) belongs to another company.")
-            root_company_msg = _lt("- Only a root company can be set on “%(record)s”. Currently set to “%(company)s”")
+            company_msg = _("- Record is company “%(company)s” and “%(field)s” (%(fname)s: %(values)s) belongs to another company.")
+            record_msg = _("- “%(record)s” belongs to company “%(company)s” and “%(field)s” (%(fname)s: %(values)s) belongs to another company.")
+            record_companies_msg = _("- “%(record)s” belongs to companies %(company)s but “%(field)s” (%(fname)s: %(values)s) belongs to another company.")
+            root_company_msg = _("- Only a root company can be set on “%(record)s”. Currently set to “%(company)s”")
             for record, name, corecords in inconsistencies[:5]:
                 if record._name == 'res.company':
-                    msg, company = company_msg, record
+                    msg, company_name = company_msg, record.display_name
                 elif record == corecords and name == 'company_id':
```

Comment:
I am not familiar with the assumptions for the `check_company` things. Here the check is `record == corecords` but the error is `Only a root company can be set` which sounds like `company.parent_id != company`. It is hard for me to understand why they are logically equivalent. Could you explain a little bit?